File: javascript/darts/darts.js
const pointMap = new Map([
  [1, 10],
  [5, 5],
  [10, 1]
]);

export const score = (x, y) => {
  let radius = [...pointMap.keys()].find(radius => coordinatesAreInsideCircle(x, y, radius));
  return radius ? pointMap.get(radius) : 0;
};

function coordinatesAreInsideCircle(x, y, radius) {
  // using the equation of a circle: x^2 + x^2 = r^2
  const xCoordinateIsInsideCircle = x**2 <= (radius**2 - y**2);
  const yCoordinateIsInsideCircle = y**2 <= (radius**2 - x**2);
  return xCoordinateIsInsideCircle && yCoordinateIsInsideCircle;
}

File: java/darts/src/main/java/Darts.java
import java.util.Map;

class Darts {

    private static final Map<Double, Integer> pointMap = Map.ofEntries(
            Map.entry(10.0, 1), // outer circle
            Map.entry(5.0, 5), // middle circle
            Map.entry(1.0, 10) // inner circle
    );

    private final double x;
    private final double y;

    Darts(double x, double y) {
        this.x = x;
        this.y = y;
    }

    int score() {
        return pointMap.keySet().stream()
                .sorted()
                .filter(this::coordinatesAreInsideCircle)
                .findFirst()
                .map(pointMap::get)
                .orElse(0);
    }

    private boolean coordinatesAreInsideCircle(double radius) {
        // using the equation of a circle: x^2 + x^2 = r^2
        boolean xCoordinateIsInsideCircle = x * x <= (radius * radius - y * y);
        boolean yCoordinateIsInsideCircle = y * y <= (radius * radius - x * x);
        return xCoordinateIsInsideCircle && yCoordinateIsInsideCircle;
    }

}
